ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py

```diff
@@ -1940,6 +1940,43 @@ def test_tensorflow_rsqrt(
         x=x[0],
     )
 
+#segment_sum
+@handle_frontend_test(
+    fn_tree="tensorflow.math.segment_sum",
+    dtype_and_data=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        shape = (5, 6),        
+    ),
+    dtype_and_segment=helpers.dtype_and_values(
+        available_dtypes=["int32", "int64"],
+        shape = (5, ),
+        min_value = 0, 
+        max_value = 4,
+    ),
+    test_with_out=st.just(False),
+)
+def test_tensorflow_segment_sum(
+    *,
+    dtype_and_data,
+    dtype_and_segment,
+    frontend,
+    test_flags,
+    fn_tree,
+    backend_fw,
+    on_device,
+):
+    data_dtype, data = dtype_and_data
+    segment_dtype, segment_ids = dtype_and_segment
+    helpers.test_frontend_function(
```

Comment:
The tests are still failing. It shows` TypeError: 'Array' object cannot be interpreted as an integer`
You don't need to send the dtyès as an array. Check out this call as an example

``` input_dtype, x = dtype_and_x
      helpers.test_frontend_function(
        input_dtypes=input_dtype,
        backend_to_test=backend_fw,
        frontend=frontend,
        test_flags=test_flags,
        fn_tree=fn_tree,
        on_device=on_device,
        x1=x[0],
        x2=x[1],
    )

```
To the kwarg `input_dtypes` you only need to pass the `data_dtype`. And you pass the inputs at the end of the function call. And so on. Let me know if this makes sense!